tortoise/backends/aioodbc/client.py

```diff
@@ -0,0 +1,277 @@
+import asyncio
+from functools import wraps
+from typing import Any, Callable, List, Optional, SupportsInt, Tuple, TypeVar, Union
+
+import aioodbc
+import pyodbc
+# from asyncpg.transaction import Transaction
+import pypika
+from pypika import OracleQuery as OrigOracleQuery
+
+
+from tortoise.backends.aioodbc.executor import AioodbcExecutor
+from tortoise.backends.aioodbc.schema_generator import AioodbcSchemaGenerator
+from tortoise.backends.base.client import (
+    BaseDBAsyncClient,
+    BaseTransactionWrapper,
+    Capabilities,
+    ConnectionWrapper,
+    NestedTransactionPooledContext,
+    PoolConnectionWrapper,
+    TransactionContext,
+    TransactionContextPooled,
+)
+from tortoise.exceptions import (
+    DBConnectionError,
+    IntegrityError,
+    OperationalError,
+    TransactionManagementError,
+)
+
+FuncType = Callable[..., Any]
+F = TypeVar("F", bound=FuncType)
+
+
+def translate_exceptions(func: F) -> F:
+    @wraps(func)
+    async def translate_exceptions_(self, *args):
+        try:
+            return await func(self, *args)
+        except pyodbc.ProgrammingError as exc:
+            raise OperationalError(exc)
+        except pyodbc.IntegrityError as exc:
+            raise IntegrityError(exc)
+        except pyodbc.Error as exc:
+            raise OperationalError(exc)
+        # except pyodbc.erras exc:  # pragma: nocoverage
+            # raise TransactionManagementError(exc)
+
+    return translate_exceptions_  # type: ignore
+
+
+
+
+
+
+from pypika import OracleQuery as OrigOracleQuery
```

Comment:
We are not using this from upstream?

Reply:
I'm not sure what you are asking?

Reply:
If by upstream you mean why am I manipulating the OracleQuery instead of asking pypika to integrate. The reason is Pypika basically refuses to make a working limit function for oracle. 
https://github.com/kayak/pypika/issues/211

As for quotechar... Specifying quotechar is their response to needing no/different quotechars.
https://github.com/kayak/pypika/issues/37

Reply:
K, requested PyPika to integrate upstream.

Reply:
Oh, interesting. So we may just have to maintain our own variant for a while.